src/ai/whylabs/templates/batch_bigquery_template.py

```diff
@@ -30,30 +43,53 @@ class TemplateArgs:
     logging_level: str
     date_column: str
     date_grouping_frequency: str
+    segment_columns: Optional[str]
 
 
-class ViewCombiner(beam.CombineFn):
-    def __init__(self, args: TemplateArgs):
-        self.logger = logging.getLogger("ViewCombiner")
-        self.logging_level = args.logging_level
+@dataclass
+class InputBigQuerySQL:
+    query: str
 
-    def setup(self) -> None:
-        self.logger.setLevel(logging.getLevelName(self.logging_level))
+
+@dataclass
+class InputBigQueryTable:
+    table_spec: str
 
     def create_accumulator(self) -> DatasetProfileView:
         return DatasetProfile().view()
 
-    def add_input(self, accumulator: DatasetProfileView, input: DatasetProfileView) -> DatasetProfileView:
-        return accumulator.merge(input)
+    def add_input(self, accumulator: DatasetProfileView, input: List[DatasetProfileView]) -> DatasetProfileView:
+        if len(input) == 0:
+            return accumulator
+
+        profile = DatasetProfile()
+        profile.track(pd.DataFrame.from_dict(input))
+        ret = accumulator.merge(profile.view())
+        return ret
 
     def merge_accumulators(self, accumulators: List[DatasetProfileView]) -> DatasetProfileView:
+        if len(accumulators) == 1:
+            # logger.info('Returning accumulator, only one to merge')
+            return accumulators[0]
+
         view: DatasetProfileView = DatasetProfile().view()
         for current_view in accumulators:
             view = view.merge(current_view)
         return view
 
-    def extract_output(self, accumulator: DatasetProfileView) -> DatasetProfileView:
-        return accumulator
+
+@dataclass
+class InputOffset:
+    offset: int
+    table_spec: str
+    timezone: tzinfo
+    query: str
+
+
+class SegmentDefinition(NamedTuple):
+    segment_column: str
+    segment: Segment
+    date_group: str
```

Comment:
Out of curiosity, do you know how this is used as a key? Does it stringify this via pickle or use the __str__?